File: BGC_WebApi/BGC_WebApi/DTOs/RpgSessionDTO.cs
using BGC_DataAccess.Entities;
using BGC_DataAccess.Enums;

namespace BGC_WebApi.DTOs;

/// <summary>
/// DTO for RPG session
/// </summary>
public class RpgSessionDTO
{
    /// <summary>
    /// Id
    /// </summary>
    public int RpgSessionId { get; set; }
    /// <summary>
    /// Name
    /// </summary>
    public string Name { get; set; }
    /// <summary>
    /// Description
    /// </summary>
    public string? Description { get; set; }
    /// <summary>
    /// Day and time of the game
    /// </summary>
    public DateTime Schedule { get; set; }
    /// <summary>
    /// Status of the session
    /// </summary>
    public SessionStatus Status { get; set; }
    /// <summary>
    /// Id of the RPG campaign
    /// </summary>
    public int RpgCampaignId { get; set; }
    /// <summary>
    /// RPG campaign
    /// </summary>
    public RpgCampaignDTO Campaign { get; set; }
    /// <summary>
    /// Id of the game table
    /// </summary>
    public int? GameTableId { get; set; }
    /// <summary>
    /// Game table
    /// </summary>
    public GameTableDTO Table { get; set; }

    //public ICollection<RpgSessionGame> RpgSessionGames { get; set; }
}

File: BGC_Portal/src/app/api/models/rpg-session-dto.ts
/* tslint:disable */
/* eslint-disable */
import { GameTableDto } from './game-table-dto';
import { RpgCampaignDto } from './rpg-campaign-dto';
import { SessionStatus } from './session-status';
export interface RpgSessionDto {
  campaign?: RpgCampaignDto;
  description?: null | string;
  gameTableId?: null | number;
  name?: null | string;
  rpgCampaignId?: number;
  rpgSessionId?: number;
  schedule?: string;
  status?: SessionStatus;
  table?: GameTableDto;
}
